model/DataObject.php

```diff
@@ -2345,6 +2345,10 @@ public function setField($fieldName, $val) {
 
 					// Value has changed as well, not just the type
 					$this->changed[$fieldName] = 2;
+
```

Comment:
Can you please describe what this code does through an inline comment? It's not very obvious :)